### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by AlexTeusz**

### CODE REVIEW

File: transformers/avayaVoiceChannel/README.MD [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Description
+Avaya Voice Channel Transformer translates the communication between Avaya CPaaS calls,sms messages and Cognigy's text in real time.
+
+# Usage
+The first step is to upload Avaya's Extension, then create a Cognigy Flow Bot with Avaya's extension nodes such as play,prompt,record,handover,hangup,conference.
```
Comment (AlexTeusz) [inline transformers/avayaVoiceChannel/README.MD:5]: Please add a detailed description of how to use the extension with this transformer. The developer needs to be able to follow a step-by-step tutorial.

File: transformers/avayaVoiceChannel/README.MD [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Description
+Avaya Voice Channel Transformer translates the communication between Avaya CPaaS calls,sms messages and Cognigy's text in real time.
```
Comment (AlexTeusz) [inline transformers/avayaVoiceChannel/README.MD:2]: What is this Avaya service and where can a developer get access and credentials for CPaaS etc.? Please add this information to the doc.

File: transformers/avayaVoiceChannel/transformer.ts [hunk 3]
```diff
@@ -0,0 +1,231 @@
+/** 
+ * Avaya voice channel
+ * version 0.1
+ */
+const COGNIGY_BASE_URL = 'https://endpoint-trial.cognigy.ai/';
```
Comment (AlexTeusz) [inline transformers/avayaVoiceChannel/transformer.ts:5]: Please add a comment why the base url is required.

File: transformers/avayaVoiceChannel/transformer.ts [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+		 * from the request. Example:
+		 *
+		 * const { userId, sessionId, text, data } = request.body;
+		 *
+		 * Note that the format of the request body will be different for
+		 * every Endpoint, and the example above needs to be adjusted
+		 * accordingly.
+		 */
+		const { body } = request as CPaaSRequest;
+		console.log('body='.concat(JSON.stringify(body)));
```
Comment (AlexTeusz) [inline transformers/avayaVoiceChannel/transformer.ts:33]: Please add a boolean DEBUG_MODE to the top configuration and only print out the console logs when the debug mode is on.

File: transformers/avayaVoiceChannel/transformer.ts [hunk 5]
```diff
@@ -0,0 +1,231 @@
+/**
```
Comment (AlexTeusz) [inline transformers/avayaVoiceChannel/transformer.ts:1]: Please format the entire code file.

File: transformers/avayaVoiceChannel/transformer.ts [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+	 * @returns The output that will be formatted into the final response in the 'handleExecutionFinished' transformer.
+	 */
+	handleOutput: async ({ output, endpoint, userId, sessionId }) => {
+		console.log('handleOutput called');
+        const sessionStorage = await getSessionStorage(userId, sessionId);
+		if (output.data != null && output.data._cognigy != null) {
+			const payload = output.data._cognigy;
+			console.log('payload='.concat(payload));
+			const activities = payload._spoken.json.activities;
+			activities.forEach( (a) => {
```
Comment (AlexTeusz) [inline transformers/avayaVoiceChannel/transformer.ts:88]: Please call it *activity* and not *a* in the forEach loop.

File: transformers/avayaVoiceChannel/transformer.ts [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+	 * @param userId The unique ID of the user.
+	 * @param sessionId The unique ID for this session. Can be used together with the userId
+	 * to retrieve the sessionStorage object.
+	 * 
+	 * @returns The output that will be formatted into the final response in the 'handleExecutionFinished' transformer.
+	 */
+	handleOutput: async ({ output, endpoint, userId, sessionId }) => {
+		console.log('handleOutput called');
+        const sessionStorage = await getSessionStorage(userId, sessionId);
+		if (output.data != null && output.data._cognigy != null) {
```
Comment (AlexTeusz) [inline transformers/avayaVoiceChannel/transformer.ts:84]: Change *!=* to *!==* for a total check.

File: transformers/avayaVoiceChannel/transformer.ts [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+	 * @param response The express response object that can be used to send a custom response back to the user.
+	 *
+	 * @returns An object that will be sent to the user, unchanged. It therefore has to have the
+	 * correct format according to the documentation of the specific Endpoint channel.
+	 */
+	handleExecutionFinished: async ({ processedOutput, outputs, userId, sessionId, endpoint, response }) => {
+		console.log("handleExecutionFinished called");
+		let url = COGNIGY_BASE_URL + endpoint.URLToken;
+		const sessionStorage = await getSessionStorage(userId, sessionId);
+		let r = 'default';
```
Comment (AlexTeusz) [inline transformers/avayaVoiceChannel/transformer.ts:163]: Please change *r* to a better variable name. What does r stand for?

File: transformers/avayaVoiceChannel/transformer.ts [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+	return (r);
+}
+
+
+
+interface CPaaSRequest {
+	body: CPaaSBody;
+}
+
+interface CPaaSBody {
```
Comment (AlexTeusz) [inline transformers/avayaVoiceChannel/transformer.ts:214]: Please move the interface to the top of the file.

File: transformers/avayaVoiceChannel/transformer.ts (hunk 3 above)
Comment (dayw168) [reply transformers/avayaVoiceChannel/transformer.ts:5]: Avaya CPaaS cmd "Gather" needs to know the url to pass the speech result to. In this case, the url is the full url of Cognigy's endpoint. It'd be better if Cognigy has API to get the current Cognigy's base url.

File: transformers/avayaVoiceChannel/README.MD (hunk 1 above)
Comment (dayw168) [reply transformers/avayaVoiceChannel/README.MD:5]: made changes per suggested

File: transformers/avayaVoiceChannel/README.MD (hunk 2 above)
Comment (dayw168) [reply transformers/avayaVoiceChannel/README.MD:2]: made the changes per suggestion

File: transformers/avayaVoiceChannel/transformer.ts (hunk 4 above)
Comment (dayw168) [reply transformers/avayaVoiceChannel/transformer.ts:33]: made the changes per suggestion.

File: transformers/avayaVoiceChannel/transformer.ts (hunk 6 above)
Comment (dayw168) [reply transformers/avayaVoiceChannel/transformer.ts:88]: made the changes per suggestion

File: transformers/avayaVoiceChannel/transformer.ts (hunk 7 above)
Comment (dayw168) [reply transformers/avayaVoiceChannel/transformer.ts:84]: it stopped working after changing != to !==. So no changes have been made on this suggestion.

File: transformers/avayaVoiceChannel/transformer.ts (hunk 8 above)
Comment (dayw168) [reply transformers/avayaVoiceChannel/transformer.ts:163]: made the changes per suggestion

File: transformers/avayaVoiceChannel/transformer.ts (hunk 9 above)
Comment (dayw168) [reply transformers/avayaVoiceChannel/transformer.ts:214]: made the changes per suggestion.

**Pull request merged**